core/src/main/java/gyro/core/validation/CollectionMinValidator.java

```diff
@@ -0,0 +1,23 @@
+package gyro.core.validation;
+
+import java.util.Collection;
+
+import gyro.core.GyroException;
+import gyro.core.resource.Diffable;
+
+public class CollectionMinValidator implements Validator<CollectionMin> {
+
+    @Override
+    public boolean isValid(Diffable diffable, CollectionMin annotation, Object value) {
+        if (value instanceof Collection) {
+            return ((Collection<?>) value).size() >= annotation.value();
+        }
+
+        throw new GyroException("Invalid usage of '@CollectionMin' validation annotation. Can only be used on types that extend Collection.");
```

Comment:
I think this should throw `gyro.util.Bug` since there is nothing end user can do to resolve this.